surelock/src/main/java/com/smashingboxes/surelock/Surelock.java

```diff
@@ -228,11 +237,15 @@ public void enrollFingerprintAndStore(String key, byte[] valueToEncrypt,
     public void enrollFingerprintAndStore(String key,
                                           SurelockFragment surelockFragment,
                                           @NonNull FragmentManager fragmentManager,
-                                          String fingerprintDialogFragmentTag) {
+                                          String fingerprintDialogFragmentTag) throws SurelockException {
         initKeyStoreKey();
         Cipher cipher;
         try {
-            cipher = initCipher(Cipher.ENCRYPT_MODE);
+            try {
+                cipher = initCipher(Cipher.ENCRYPT_MODE);
+            } catch (InvalidKeyException | UnrecoverableKeyException | KeyStoreException e) {
+                throw new SurelockException("Failed to init Cipher for encryption", e);
```

Comment:
Why are the exceptions here `SurelockException`s instead of `SurelockInvalidKeyException`s? Is it because at this point, the fingerprints haven't been enrolled so the store hasn't been invalidated yet?